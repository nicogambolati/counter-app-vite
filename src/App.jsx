import PropTypes from 'prop-types';

// const newMessage = {
//   message: 'Hello World',
//   title: 'Nicolas'
// };

const getResult = (a, b) => {
  return a + b
}

function App ({title, subtitle}) {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subtitle}</h2>
      <h2>{subtitle}</h2>
      {/* <code>{ JSON.stringify(newMessage) }</code> */}
      {/* <h2>{getResult(1,2)}</h2> */}
      <p>Soy un subtitulo</p>
    </>
  )
}

export default App

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

App.defaultProps = {
  title: 'No hay titulo',
  subtitle: 'NO hay subtitulo',
  name: 'Nicolas'
}