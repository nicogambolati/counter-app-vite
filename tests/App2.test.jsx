import { render, screen } from "@testing-library/react";
import App from "../src/App";

describe("prueba de <App>", () => {
  const title = "Hola, soy Goku";
  const subtitle = "subtitulo de prueba";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<App title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar el mensage "Hola, soy Goku"', () => {
    render(<App title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe el titulo en un h1", () => {
    render(<App title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(title);
  });

  test('debe el subtitulo enviado por props', () => { 
    render(<App title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
   })
});
