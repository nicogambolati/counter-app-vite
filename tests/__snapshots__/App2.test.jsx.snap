// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba de <App> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <h2>
    NO hay subtitulo
  </h2>
  <h2>
    NO hay subtitulo
  </h2>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
